File: robertly-ui/src/app/models/sync.ts
import { ExerciseLog } from "./exercise-log.model";

export interface ExerciseLogsSyncPushRequest {
  creates: ExerciseLog[];
  updates: ExerciseLog[];
  deletes: number[];
  seriesIdsToDelete: number[];
}

export interface ExerciseLogsSyncPullResponse {
  exerciseLogs: ExerciseLog[];
  serverTimeUtc: string;
}

File: robertly-net-api/api/Models/Sync.cs
using System;
using System.Collections;
using System.Collections.Generic;

namespace robertly.Models;

public class ExerciseLogsSyncPushRequest
{
  public IEnumerable<Models.ExerciseLog> Creates { get; set; } = [];
  public IEnumerable<Models.ExerciseLog> Updates { get; set; } = [];
  public IEnumerable<int> Deletes { get; set; } = [];
  public IEnumerable<int> SeriesIdsToDelete { get; set; } = [];
}

public class ExerciselogsSyncPullResponse
{
  public IEnumerable<Models.ExerciseLog> ExerciseLogs { get; set; } = [];
  public DateTime ServerTimeUtc { get; set; }
}
